http/openapi.go

```diff
@@ -111,7 +111,7 @@ func NewOpenAPISpec() (*openapi3.T, error) {
 		Servers: openapi3.Servers{
 			&openapi3.Server{
 				Description: "Local DefraDB instance",
-				URL:         "http://localhost:9181/api/v0",
+				URL:         "http://127.0.0.1:9181/api/v0",
```

Comment:
todo: This indeed solves the problem in the case of using the playground locally but in the case where someone wants to expose this endpoint using a different port or a public address, it will be incorrect. I do think that this can be used as a placeholder or documentation for the openAPI server within the codebase but ideally the URL would be assigned based on the configured DefraDB server address.

Unfortunately the openAPI object definition is created within the handler instantiation which doesn't have access to the server options that contain the server address. What can be done though is to pass the server address to the handler function for the openAPI endpoint using the request context. I've created a rough draft of what that might look like here: https://github.com/sourcenetwork/defradb/commit/fd640e2968a92f78796d3b81d9882b151392a393

I'm happy to get on a call to discuss this further if needed 🙂

Reply:
I have incorporated these changes, but I have addressed something that Fred and I discussed.

The proposed changes he drafted up contained one bug, in that the URL was treated as a relative path instead of an absolute path. This resulted in URLs of the form: `http://127.0.0.1/127.0.0.1/api/v0`

My proposed fix is to prepend the URL with either HTTP or HTTPS, pending an `isTLS` flag which is set by examining the `options` parameter that is passed to the `NewServer` function of `server.go`

Reply:
Just wanted to add a thought. If you use the http `Host` header from the request it would allow it to work even when behind a reverse proxy, load balancer, or different DNS name.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host

Reply:
In light of this discussion, I have rolled back the changes. The PR is now two line changes, one in `openapi.go` and one in the docs.